mavros/src/plugins/rc_io.cpp

```diff
@@ -247,6 +247,18 @@ class RCIOPlugin : public plugin::PluginBase {
 		ovr.chan6_raw = req->channels[5];
 		ovr.chan7_raw = req->channels[6];
 		ovr.chan8_raw = req->channels[7];
+		if (m_uas->fcu_link->get_protocol_version() == mavconn::Protocol::V20) {
+			ovr.chan9_raw = req->channels[8];
+			ovr.chan10_raw = req->channels[9];
+			ovr.chan11_raw = req->channels[10];
+			ovr.chan12_raw = req->channels[11];
+			ovr.chan13_raw = req->channels[12];
+			ovr.chan14_raw = req->channels[13];
+			ovr.chan15_raw = req->channels[14];
+			ovr.chan16_raw = req->channels[15];
+			ovr.chan17_raw = req->channels[16];
+			ovr.chan18_raw = req->channels[17];
+		}
```

Comment:
1. That code should be generated by cog.
2. With change in message type you'll switching from std::array to std::vector (or boost:: for ros1), so you must verify size of `channels`.